Cython/Compiler/ExprNodes.py

```diff
@@ -14823,11 +14823,16 @@ def generate_result_code(self, code):
             return
         test_func = self._special_builtins.get(self.arg.type)
         if test_func is not None:
-            checks = ["(%s != Py_None)" % self.arg.py_result()] if self.arg.may_be_none() else []
-            checks.append("(%s(%s) != 0)" % (test_func, self.arg.py_result()))
-            code.putln("%s = %s;" % (self.result(), '&&'.join(checks)))
+            if self.arg.may_be_none():
+                code.putln(f"if ({self.arg.py_result()} == Py_None) {self.result()} = 0;")
+                code.putln("else")
+            code.putln("{")
+            code.putln(f"Py_ssize_t {Naming.quick_temp_cname} = {test_func}({self.arg.py_result()});")
```

Comment:
Unicode strings have an actual test function here (rather than returning the plain length), which, incidentally, suffers from the same problem in the Limited API case.

Given that `bint` fits into a `Py_ssize_t`, the temp variable type here should be fine. Might be worth a comment on the function dict above, though, as a reminder that we're expecting something int-ish here. (And the existing comment needs changing to `…_SAFE_SIZE`.)